ndcube/ndcube.py

```diff
@@ -1330,6 +1330,50 @@ def squeeze(self, axis=None):
         return self[tuple(item)]
 
 
+def fill(self, fill_value, unmask=False, uncertainty_fill_value=None, fill_in_place=False):
+    """
+    Replaces masked data values with input value.
+
+    Returns a new instance or alters values in place.
+
+    Parameters
+    ----------
+    fill_value: `numbers.Number` or scalar `astropy.unit.Quantity`
+        The value to replace masked data with.
+    unmask: `bool`, optional
+        If True, the newly filled masked values are unmasked. If False, they remain masked
+        Default=False
+    uncertainty_fill_value: `numbers.Number` or scalar `astropy.unit.Quantity`, optional
+        The value to replace masked uncertainties with.
+    fill_in_place: `bool`, optional
+        If `True`, the masked values are filled in place.  If `False`, a new instance is returned
+        with masked values filled.  Default=False.
+    """
+    # ...code implementation here.
+    kwargs = {}
+    kwargs["data"] = self.data
+    kwargs["mask"] = self.mask
+    kwargs["uncertainty"] = self.uncertainty
```

Comment:
We only need to generate kwargs if we want to create a new instance, i.e. if `fill_in_place=False`.  So let's set the data, mask and uncertainty to variables that can be pointers to the exisiting cube's attributes, or deepcopies of them, depending on `fill_in_place`.  So something like:
```suggestion
    if fill_in_place:
        new_data = self.data
        new_uncertainty = self.uncertainty
        new_mask = False if unmask else self.mask
    else:
        new_data = copy.deepcopy(self.data)
        new_uncertainty = copy.deepcopy(self.uncertainty)
        new_mask = False if unmask else copy.deepcopy(self.mask)
```
Variable references below need to be changed accordingly